visvis2/_wrappers.py

```diff
@@ -0,0 +1,202 @@
+import ctypes
+
+import wgpu
+import numpy as np
+
+
+def array_from_shader_type(spirv_type):
+    """ Get a numpy array object from a SpirV type from python-shader.
+    """
+    return np.asarray(spirv_type())
+
+
+# todo: can this be generic enough, keeping the GPU bits out / optional?
+
+
+class BaseBufferWrapper:
```

Comment:
do you think it's possible to move the WGPU specific stuff into the WgpuRenderer and make it possible for an OpenGL renderer to also support BufferWrapper?